common/lib/common-definitions/src/logger.ts

```diff
@@ -15,23 +15,37 @@ export type TelemetryEventCategory = "generic" | "error" | "performance";
  * easily add fields to objects that shouldn't be logged and not realize it's going to be logged.
  * General best practice is to explicitly log the fields you care about from objects.
  */
-export type TelemetryEventPropertyType = string | number | boolean | undefined;
+export type TelemetryEventPropertyType =
+    | string
+    | number
+    | boolean
+    | undefined
+    | (string | number | boolean)[];
 
 /**
  * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
  * to mark pieces of information that should be organized or handled differently by loggers in various first or third
  * party scenarios. For example, tags are used to mark PII that should not be stored in logs.
  */
-export interface ITaggedTelemetryPropertyType {
-    value: TelemetryEventPropertyType;
-    tag: string;
+ export interface TaggedProperty<TValue> {
```

Comment:
maybe ITaggedTelemetryProperty? It's interface, and we start all interfaces with I.